.eslintrc.js

```diff
@@ -1,46 +1,299 @@
+  '@typescript-eslint/prefer-optional-chain': 'error',
+  '@typescript-eslint/prefer-readonly': 'error',
+  '@typescript-eslint/prefer-reduce-type-parameter': 'error',
+  '@typescript-eslint/prefer-string-starts-ends-with': 'error',
+  '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
+  '@typescript-eslint/require-array-sort-compare': ['error', { ignoreStringArrays: true }],
+  '@typescript-eslint/return-await': 'error',
+  '@typescript-eslint/semi': ['error', 'always'],
+  '@typescript-eslint/space-infix-ops': 'error',
+  '@typescript-eslint/switch-exhaustiveness-check': 'error',
+  '@typescript-eslint/type-annotation-spacing': 'error',
+  '@typescript-eslint/unified-signatures': 'error',
+};
+
+/**
+ * A lot of import rules are already covered by the Typescript compiler, so we don't use any
+ * of the eslint-plugin-import presets. These rules are helpful additions.
+ * https://github.com/benmosher/eslint-plugin-import/tree/master/docs/rules
+ */
+const importRules = {
+  'import/first': 'error',
+  'import/newline-after-import': 'error',
+  'import/no-absolute-path': 'error',
+  'import/no-amd': 'error',
+  'import/no-anonymous-default-export': 'error',
+  'import/no-commonjs': 'error',
+  'import/no-cycle': 'error',
+  // preferred over Typescript ESLint's and ESLint's no-duplicate-imports rules
+  'import/no-duplicates': 'error',
+  'import/no-dynamic-require': 'error',
+  'import/no-extraneous-dependencies': 'error',
+  'import/no-mutable-exports': 'error',
+  'import/no-named-as-default': 'error',
+  'import/no-named-default': 'error',
+  'import/no-self-import': 'error',
+  'import/no-unassigned-import': [
+    'error',
+    {
+      allow: [
+        '**/*.scss',
+        '**/*.css',
+        'jest', // can probably remove because an explicit import is not required in Jest context
+        '@testing-library/jest-dom/extend-expect',
+      ],
+    },
+  ],
+  'import/no-useless-path-segments': 'error',
+  'import/no-webpack-loader-syntax': 'error',
+  'import/order': 'error', // preferred over ESLint's sorted-imports rule
 };
 
 module.exports = {
   root: true,
   parser: '@typescript-eslint/parser',
   parserOptions: {
+    ecmaVersion: 2020,
     tsconfigRootDir: __dirname,
-    project: [
-      './tsconfig.json',
-      './tsconfig.test.json',
-    ],
+    project: ['./tsconfig.json', './tsconfig.test.json'],
   },
-  plugins: [
-    '@typescript-eslint',
-  ],
+  plugins: ['@typescript-eslint', 'import', 'prettier', 'prefer-arrow'],
   extends: [
-    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
     'plugin:@typescript-eslint/recommended',
     'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'prettier',
   ],
   env: {
-    'node': true,
+    node: true,
   },
   rules: {
-    ...rules,
+    ...coreESLintRules,
+    ...typescriptESLintRules,
+    ...importRules,
+    'prefer-arrow/prefer-arrow-functions': 'error',
```

Comment:
Can this go in the coreESLintRules section?

Reply:
Technically it would still work if we put it there, but because the rule is coming from a seperate plugin, it was included on it's own here.